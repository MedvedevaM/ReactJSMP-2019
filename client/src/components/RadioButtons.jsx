import React from "react";

export const RadioButtons = props => {
    const { parameters, searchBy } = props;
    const formattedInputs = parameters.map(parameter => (
      <React.Fragment key={`containerOf${parameter}SortParameter`}>
        <input
            id={parameter}
            type="radio"
            name="parameters"
            key={parameter}
            value={parameter}
            defaultChecked={searchBy === parameter}
        />
        <label className="uppercase bold" htmlFor={parameter}>{parameter}</label>
      </React.Fragment>
    ));
    return (
      [formattedInputs]
    );
};
