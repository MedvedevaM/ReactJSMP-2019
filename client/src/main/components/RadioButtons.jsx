import React from 'react';

const RadioButtons = (props) => {
  const { parameters, searchBy, onClick } = props;
  const formattedInputs = parameters.map(parameter => (
    <React.Fragment key={`containerOf${parameter}SortParameter`}>
      <input
        data-parameter={parameter}
        type="radio"
        name="parameters"
        key={parameter}
        value={parameter}
        defaultChecked={searchBy === parameter}
        onClick={onClick}
      />
      <label data-parameter={parameter} className="uppercase bold" htmlFor={parameter}>{parameter}</label>
    </React.Fragment>
  ));
  return (
    [formattedInputs]
  );
};

export default RadioButtons;
