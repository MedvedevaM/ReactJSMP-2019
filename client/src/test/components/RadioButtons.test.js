import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json';
import { RadioButtons } from "../../main/components/RadioButtons.jsx";

configure({ adapter: new Adapter() });

const parameters = ["first_parameter", "second_parameter", "third_parameter"];

it("check radio buttons rendering", () => {
    const radioButtons = shallow( <RadioButtons parameters={parameters} searchBy="second_parameter" /> );
    expect(toJson(radioButtons)).toMatchSnapshot();
    expect(radioButtons.find('input').length).toBe(3);
});

