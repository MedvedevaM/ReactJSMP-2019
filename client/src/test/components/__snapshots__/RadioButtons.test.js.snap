// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check radio buttons rendering 1`] = `
Array [
  <Fragment
    key="containerOffirst_parameterSortParameter"
  >
    <input
      data-parameter="first_parameter"
      defaultChecked={false}
      key="first_parameter"
      name="parameters"
      type="radio"
      value="first_parameter"
    />
    <label
      className="uppercase bold"
      htmlFor="first_parameter"
    >
      first_parameter
    </label>
  </Fragment>,
  <Fragment
    key="containerOfsecond_parameterSortParameter"
  >
    <input
      data-parameter="second_parameter"
      defaultChecked={true}
      key="second_parameter"
      name="parameters"
      type="radio"
      value="second_parameter"
    />
    <label
      className="uppercase bold"
      htmlFor="second_parameter"
    >
      second_parameter
    </label>
  </Fragment>,
  <Fragment
    key="containerOfthird_parameterSortParameter"
  >
    <input
      data-parameter="third_parameter"
      defaultChecked={false}
      key="third_parameter"
      name="parameters"
      type="radio"
      value="third_parameter"
    />
    <label
      className="uppercase bold"
      htmlFor="third_parameter"
    >
      third_parameter
    </label>
  </Fragment>,
]
`;
